src/watch/watch-proxy.ts

```diff
@@ -17,6 +22,23 @@ export default function watch(configs: RollupOptions[] | RollupOptions): RollupW
 	return emitter;
 }
 
+function checkWatchConfig(config: MergedRollupOptions[]): boolean {
+	for (const item of config) {
+		if (item.input && item.output) {
+			const input = typeof item.input === 'string' ? ensureArray(item.input) : item.input;
+			const output = ensureArray(item.output);
+			for (const index in input) {
+				const inputPath = input[index as keyof typeof input] as string;
+				const subPath = output.some(o => o.dir && inputPath.startsWith?.(o.dir));
```

Comment:
While only an edge case, "startsWith" can have false positives, e.g. if the output dir is `foo/website/` and the input is `foo/website.js`, which feels slightly plausible to me. AI suggest something like

```js
function isSubPath(parentDir, filePath) {
  const relative = path.relative(parentDir, filePath);
  return relative && !relative.startsWith('..') && !path.isAbsolute(relative);
}
```

which does not feel too bad to me, though of course it is more involved. Alternatively, one could probably extend "startsWith" with a check that the remaining part of inputPath starts with a path separator.

Reply:
Thanks for the review, I will change it.